### ISSUE TITLE
[WebAPI] Midi Support added

### ISSUE THREAD

**Issue by hamishwillee** [mdn/content#12792]
## Acceptance Criteria

- [x] The listed features are documented sufficiently on MDN
- [x] BCD is updated
- [x] Interactive example and data repos are updated if appropriate
- [ ] The content has been reviewed as needed

### For folks helping with Firefox related documentation

- [x] Set bugs to `dev-doc-complete`
- [ ] Add entry to Firefox release notes if feature is enabled in release
or
- [ ] Add entry to Firefox experimental features page if feature is not yet enabled in release

## Related Gecko bugs

- FF97 [1748647 ](https://bugzilla.mozilla.org/show_bug.cgi?id=1748647) - MIDI messages are delivered twice if the same port is opened twice
- FF98 [Implement the WebMIDI API](https://bugzilla.mozilla.org/show_bug.cgi?id=836897)
- FF99 [Enable WebMIDI](https://bugzilla.mozilla.org/show_bug.cgi?id=1752906) (ship bug)

**Comment by hamishwillee** [comment]
@Rumyra I've converted the WebMidi project notes into this issue. Right now my testing shows that it works on FF60 on windows, which is a bit odd because the implication of the thread is that support should appear in FF97. Anyway, as a result I'm querying what exactly is delivered in https://bugzilla.mozilla.org/show_bug.cgi?id=836897#c117.

Looking at this there are already some docs here: https://developer.mozilla.org/en-US/docs/Web/API/Web_MIDI_API. I expect the task here would be
- [ ] BCD - add entry once work out what has been "really" done in what versions.
- [ ] Docs
   - [ ] Check spec against documented API to work out if it is complete - update as needed
   - [ ] Update with a live example (makes testing easier). 
   - [ ] Maybe add link to this page, which makes a nice test tool: https://blog.karimratib.me/demos/sheetplayer/ 
- [ ] Release notes - None, it isn't released.
- [ ] Experimental features - update once we know what BCD is doing.
- [ ] dev-docs-complete - TBD

**Comment by hamishwillee** [comment]
PS, did you want to do this issue? I'm happy to!

**Comment by hamishwillee** [comment]
@Rumyra OK, BCD for WebMidi for FF97 sorted in https://github.com/mdn/browser-compat-data/pull/14874 - essentially we are considering to have been added "new" on behind a preference (ignoring previous dummy implementation) on FF. FF for android is still false, because a) preferences not set on android, b)there is no android backend.

So in FF98 the plan is that the preference will be removed. When that happens I intend to add a new entry for FF removing the preference. FF for Android will remain false until there is a backend.

Please ping me if you note that this has shipped in FF98 so I can do the next bit.

**Comment by Rumyra** [comment]
OK @hamishwillee thank you 👍  I'm going to move this to 'In Progress' as I plan to start work on the docs tomorrow

**Comment by Rumyra** [comment]
NB it looks as though this will be rolled into FF99 - I plan to get the docs work done now, but I'll keep my eye on it release info 👍

**Comment by hamishwillee** [comment]
@Rumyra Excellent. BCD update in https://github.com/mdn/browser-compat-data/pull/15152.

Excellent to get docs done and close this. Then we can start fresh on the release in (hopefully) FF99

**Comment by hamishwillee** [comment]
@Rumyra Further BCD update in FF99: https://bugzilla.mozilla.org/show_bug.cgi?id=1757153

**Comment by Rumyra** [comment]
Also https://bugzilla.mozilla.org/show_bug.cgi?id=1752906

**Comment by hamishwillee** [comment]
FYI @Rumyra 
- I've done FF99 BCD work for the release and for adding secure context.
- Updated the docs a bit for secure context: https://github.com/mdn/content/pull/13479
- NOT yet removed experimental tagging from docs (but have in BCD). Figure we can wait on BCD before doing this.
- NOT done release note and removed from experimental features.

**Comment by Rumyra** [comment]
Thanks @hamishwillee :) - I've only just created a pr for the edits to 98 release notes and updating 99 & 100 - so it's probably best to wait for that to go through before updating anyway 👍

**Comment by hamishwillee** [comment]
Cool. I'm leaving to you, but I'll track the BCD stuff.

**Comment by hamishwillee** [comment]
@Rumyra Just FYI, and you might already have known this, but access to web midi is gated on a site permission add on. This is something I had not heard of. Some discussion here: https://bugzilla.mozilla.org/show_bug.cgi?id=1752906

**Comment by Rumyra** [comment]
Yes I am - thanks @hamishwillee - I think I'll tackle the events pages first (move to new format) but I want to update the api landing page - which I think will be a good page to explain this 👍

**Comment by hamishwillee** [comment]
@Rumyra You were right https://bugzilla.mozilla.org/show_bug.cgi?id=1752906#c30 - FF100. 

BCD updated -https://github.com/mdn/browser-compat-data/pull/15610 - I am assuming you will let me know if anything more is required of me on this one :-)

**Comment by Rumyra** [comment]
Amazing @hamishwillee - that was precisely the thing that was needed. Luckily I hadn't done the rel note yet so I'll just push the few last tasks for this to this coming month 👍

**Comment by hamishwillee** [comment]
Yes, BCD fixed in https://github.com/mdn/browser-compat-data/pull/16062

**Comment by Rumyra** [comment]
Star @hamishwillee thank you!

**Comment by Rumyra** [comment]
I'm going to add a note to the Experiemental Features page which will mean the work on this is primarily done - we can add a release note when it comes out of preview 👍

**Comment by Rumyra** [comment]
Still on hold so closing and will re-open once it comes around again 👍
